HetsAPI/Python.hs

```diff
@@ -86,49 +223,148 @@ instance Show PyProver where
+getAvailableComorphisms :: PyTheory -> [PyComorphism]
+getAvailableComorphisms (PyTheory theory) = PyComorphism <$> HP.getAvailableComorphisms theory
+
+getUsableConsistencyCheckers :: PyTheory -> IO [(PyConsChecker, PyComorphism)]
+getUsableConsistencyCheckers (PyTheory th) =
+    fmap (bimap PyConsChecker PyComorphism) <$> HP.getUsableConsistencyCheckers th
+
+checkConsistency :: HDT.TheoryPointer -> PyConsCheckingOptions -> IO ConsistencyStatus
+checkConsistency ptr = HP.checkConsistency ptr . toConsCheckingOptions
+
+checkConsistencyAndRecord :: HDT.TheoryPointer -> PyConsCheckingOptions -> IO (ConsistencyStatus, LibEnv)
+checkConsistencyAndRecord ptr = HP.checkConsistencyAndRecord ptr . toConsCheckingOptions
+
+getAllSentences :: PyTheory -> SentenceByName
+getAllSentences (PyTheory theory) = HI.getAllSentences theory
+
+getAllAxioms :: PyTheory -> SentenceByName
+getAllAxioms (PyTheory theory) = HI.getAllAxioms theory
+
+getAllGoals :: PyTheory -> SentenceByName
+getAllGoals (PyTheory theory) = HI.getAllGoals theory
+
+getProvenGoals :: PyTheory -> SentenceByName
+getProvenGoals (PyTheory theory) = HI.getProvenGoals theory
+
+getUnprovenGoals :: PyTheory -> SentenceByName
+getUnprovenGoals (PyTheory theory) = HI.getUnprovenGoals theory
+
+prettySentence :: PyTheory -> Sentence -> String
+prettySentence (PyTheory theory) = HI.prettySentenceOfTheory theory
+
+signatureOfTheory :: PyTheory -> ExtSign HDT.SignatureJSON HDT.SymbolJSON
+signatureOfTheory (PyTheory GT.G_theory { GT.gTheorySign = sig }) = ExtSign {
+        plainSign = encode (plainSign sig),
+        nonImportedSymbols = Set.map encode $ nonImportedSymbols sig
+    }
+
+logicNameOfTheory :: PyTheory -> String
+logicNameOfTheory (PyTheory GT.G_theory { GT.gTheoryLogic = lid } ) = language_name lid
+
+logicDescriptionOfTheory :: PyTheory -> String
+logicDescriptionOfTheory (PyTheory GT.G_theory { GT.gTheoryLogic = lid } ) = description lid
+
+getDGNodeById :: DGraph -> Int -> Maybe DGNodeLab
+getDGNodeById = lab . dgBody
+
+globalTheory :: DGNodeLab -> Maybe PyTheory
+globalTheory = fmap PyTheory . Static.DevGraph.globalTheory
+
+gmorphismOfEdge :: DGLinkLab -> PyGMorphism
+gmorphismOfEdge = PyGMorphism . Static.DevGraph.dgl_morphism
+
+{-------------------------------------------------------------------------------
+GMorphism Wrapper
+-------------------------------------------------------------------------------}
+comorphismOfGMorphism :: PyGMorphism -> PyComorphism
+comorphismOfGMorphism (PyGMorphism (GMorphism {gMorphismComor = cid})) = PyComorphism (Comorphism cid)
+
+signatureOfGMorphism :: PyGMorphism -> ExtSign HDT.SignatureJSON HDT.SymbolJSON
+signatureOfGMorphism (PyGMorphism (GMorphism {gMorphismSign = sig})) = ExtSign {
+        plainSign = encode (plainSign sig),
+        nonImportedSymbols = Set.map encode $ nonImportedSymbols sig
+    }
+
+logicNameOfGMorphism :: PyGMorphism -> String
```

Comment:
this is not the logic name, but the comorphism name

Reply:
Done, changed to `comorphismNameOfGMorphism` and `comorphismDescriptionOfGMorphism`.